tests/metrics/test_disability_observer.py

```diff
@@ -194,21 +196,43 @@ def test_disability_accumulation(
             rtol=0.0000001,
         ).all()
 
-    results_out = simulation.get_value("metrics")(pop.index)
+    # Test that metrics are saved out correctly
+    simulation.finalize()
+    simulation.report()
+    results_files = list(results_dir.rglob("*.csv"))
+    assert set(file.name for file in results_files) == set(["ylds.csv"])
+    results = pd.read_csv(results_files[0])
+
+    # yld_masks format: {cause: (filter, dw_pipeline)}
+    yld_masks = {
+        "all_causes": (slice(None), disability_weight),
+        "sick_cause_0": (pop["model_0"] == "sick_cause_0", disability_weight_0),
+        "sick_cause_1": (pop["model_1"] == "sick_cause_1", disability_weight_1),
+    }
 
-    # Check that all expected observation labels are there
-    for label in yld_stratification_mask.keys():
-        assert label in results_out.keys()
+    # Check that all expected observations are there
+    assert set(zip(results["sex"], results["cause"])) == set(
+        itertools.product(*[["Female", "Male"], list(yld_masks)])
+    )
+
+    # Check other columns
+    assert set(results.columns) == set(
+        ["sex", "cause", "measure", "random_seed", "input_draw", "value"]
+    )
+    assert (results["measure"] == "ylds").all()
+    assert (results["random_seed"] == 0).all()
+    assert (results["input_draw"]).isna().all()
```

Comment:
We had a short discussion about this at mid-day but want to confirm that we want this be to NA